pkg/operator/controller/ingress/status.go

```diff
@@ -197,13 +197,13 @@ func computeLoadBalancerStatus(ic *operatorv1.IngressController, service *corev1
 		message := "The LoadBalancer service is pending"
 
 		// Try and find a more specific reason for for the pending status.
-		createFailedReason := "CreatingLoadBalancerFailed"
+		createFailedReason := "SyncLoadBalancerFailed"
 		failedLoadBalancerEvents := getEventsByReason(operandEvents, "service-controller", createFailedReason)
 		for _, event := range failedLoadBalancerEvents {
 			involved := event.InvolvedObject
-			if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name {
-				reason = "CreatingLoadBalancerFailed"
-				message = fmt.Sprintf("The %s component is reporting CreatingLoadBalancerFailed events like: %s\n%s",
+			if involved.Kind == "Service" && involved.Namespace == service.Namespace && involved.Name == service.Name && involved.UID == service.UID {
```

Comment:
Please modify the unit tests to verify this change.  I suggest adding a `uid` parameter to `pendingLBService` and to `failedCreateLBEvent`: https://github.com/openshift/cluster-ingress-operator/blob/d7a8e3225007bcb0874f7ce440c607fac4ca9fc0/pkg/operator/controller/ingress/status_test.go#L35 https://github.com/openshift/cluster-ingress-operator/blob/d7a8e3225007bcb0874f7ce440c607fac4ca9fc0/pkg/operator/controller/ingress/status_test.go#L70
Then modify the "no events for current lb" test case: https://github.com/openshift/cluster-ingress-operator/blob/d7a8e3225007bcb0874f7ce440c607fac4ca9fc0/pkg/operator/controller/ingress/status_test.go#L126-L137
Use `service: pendingLBService("default", 1)`, and add `failedCreateLBEvent("default", 2)` to `events` so that the test case verifies that `computeLoadBalancerStatus` ignores events where the name matches but the UID does not match.